backend/src/main/java/me/performancereservation/domain/performance/repository/PerformanceScheduleRepository.java

```diff
@@ -1,7 +1,24 @@
 package me.performancereservation.domain.performance.repository;
 
 import me.performancereservation.domain.performance.entities.PerformanceSchedule;
+import me.performancereservation.domain.performance.model.SchedulePerformanceInfo;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.util.Optional;
 
 public interface PerformanceScheduleRepository extends JpaRepository<PerformanceSchedule, Long> {
-} 
\ No newline at end of file
+
+    // scheduleId로 DTO 프로젝션을 이용해서 SchedulePerformanceInfo 데이터 모델을 조회
+    @Query("""
+        SELECT new me.performancereservation.domain.performance.model.SchedulePerformanceInfo(
+            p.id, p.title, p.venue, p.price,
+            ps.id, ps.startTime, ps.endTime
+        )
+        FROM PerformanceSchedule ps
+        JOIN Performance p ON ps.performanceId = p.id
+        WHERE ps.id = :scheduleId
+    """)
+    Optional<SchedulePerformanceInfo> findSchedulePerformanceInfoByScheduleId(@Param("scheduleId") Long scheduleId);
```

Comment:
저렇게 dto로 반환 받을 때 select 절에 패키지 경로가 들어가서 좀 지저분한 느낌이 있잖아요? 저희 팀 컨벤션과는 맞지 않기는 한데 dto를 인터페이스로 사용하는 방법이 있더라구요
```suggestion
dto
public interface SchedulePerformanceInfo {
    Long getPerformanceId();
    String getTitle();
    String getVenue();
    int getPrice();
    Long getScheduleId();
    LocalDateTime getStartTime();
    LocalDateTime getEndTime();
}

@Query("""
    SELECT 
        p.id, p.title, p.venue, p.price,
        ps.id, ps.startTime, ps.endTime
    FROM PerformanceSchedule ps
    JOIN Performance p ON ps.performanceId = p.id
    WHERE ps.id = :scheduleId
""")
Optional<SchedulePerformanceInfo> findSchedulePerformanceInfoByScheduleId(@Param("scheduleId") Long scheduleId);
```

이런식으로 dto 인터페이스에는 getter 방식으로 이름을 입력하고 select 절과 인터페이스 매핑 순서를 맞추면 알아서 스프링에서 인터페이스를 구현한 프록시 객체를 만들어 반환해준다고 합니다. select 절에 dto 패키지 경로가 안 들어가니 코드를 좀 더 깔끔하게 쓸 수 있을 거 같긴 해요. 기존 dto랑 JSON 응답에도 차이가 없다고 하고요!

Reply:
신기하네요 ㄷㄷ.... 다만 저는 프로젝션의 지저분함을 좀 감수하겠습니다... 취향차이긴한데 불편해요..🥺